From 44d81d4b24f7da2a7afe806786d70d74cb329e6d Mon Sep 17 00:00:00 2001
From: Min He <min.he@intel.com>
Date: Tue, 23 Apr 2019 03:36:36 +0000
Subject: [PATCH 1395/1548] drm/i915/gvt: disable the preemption timeout in
 GVT-g guest

In GVT-g guest, preemption is disabled. However, the force preemption
related patches will trigger force GPU reset, which impacts the guest
QoS. So disabled it in this patch for GVT-g guests.

Tracked-On: projectacrn/acrn-hypervisor#3009
Signed-off-by: Min He <min.he@intel.com>
Reviewed-by: Zhao Yakui <yakui.zhao@intel.com>
---
 drivers/gpu/drm/i915/intel_lrc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/i915/intel_lrc.c b/drivers/gpu/drm/i915/intel_lrc.c
index ba63a6384e93..bc2fdc7d2f91 100644
--- a/drivers/gpu/drm/i915/intel_lrc.c
+++ b/drivers/gpu/drm/i915/intel_lrc.c
@@ -1231,6 +1231,7 @@ static void __update_queue(struct intel_engine_cs *engine,
 
 	/* Set a timer to force preemption vs hostile userspace */
 	if (timeout &&
+	    !intel_vgpu_active(engine->i915) &&
 	    __execlists_need_preempt(prio, execlists->queue_priority)) {
 		GEM_TRACE("%s preempt timeout=%uns\n", engine->name, timeout);
 
-- 
2.20.1

